delayrecorder.py

```diff
@@ -228,35 +225,25 @@ def plot_results(measurements, stats, fig_name):
         bbox=props,
     )
 
-    plt.savefig(fig_name)
-    print(f"Saved histogram to {fig_name}")
+    plt.savefig(png_file)
+    print(f"Saved histogram to {png_file}")
     plt.show()
 
 
-def main():
+def main() -> None:
     # Set up signal handler to handle keyboard interrupts
     signal.signal(signal.SIGINT, sigint_handler)
 
     args = parse_arguments()
-    if args.filename.suffix != ".csv":
-        print("Error: Provided filename is invalid or does not have .csv extension")
-        sys.exit()
-
-    # Either write the data to a CSV file or read from a preexisting one
-    if not args.readcsv:
-        g2g_delays = get_measurements_serial(args.num_measurements, args.quiet)
-        time.sleep(0.1)
-        # this function returns g2g_delays as a numpy array
-        g2g_delays, stats = generate_stats(g2g_delays)
-        write_measurements_csv(args.filename, g2g_delays, stats)
+
+    if args.readcsv:
```

Comment:
Yeah this is fine, I had put it the other way because I was putting the more commonly expected behaviour first but maybe this is more readable

Reply:
I learned this guidline:

> try to keep your if conditions simple. If you can avoid a `not`, do it.